### PR TITLE
Remove DisposedException

### PR DESCRIPTION
**Pull request by trowski**
This removes the need to wrap `Queue::push()` in try/catch if the iterator is disposed. If the producer cares if the iterator is disposed, they can check with `Queue::isDisposed()`, but often piping into a blackhole is acceptable (e.g., Pushing websocket frame data into a message stream. The data must be consumed anyway, and it doesn't matter if no one is listening on the other end).

If an iterable is used as the source, iteration stops once the iterator is disposed.

I have verified the behavior here works well in the postgres and websocket libs, which make heavy use of this library.

### CODE REVIEW

**Comment by trowski** [comment]
Closing after discussing with @kelunik. Throwing an exception from `Queue::push()` is the "safe-by-default" way for infinite producers. Such producers would need to check `Queue::isDiposed()` after each push to avoid creating more data than necessary, whereas throwing an exception cannot be ignored/forgotten. Consuming websocket frames is likely an exceptional use-case where the producer *needs* to keep consuming data from the stream.

TL;DR: There's no perfect solution, but an exception from `Queue::push()` prevents infinite loops or producing unnecessary data.
